### ISSUE TITLE
Dropdown to select a stylesheet file for the syntax highlighting in a code content element

### ISSUE THREAD

**Issue by CMSworker** [contao/contao#7243]
### Description

**/vendor/scrivo/highlight.php** (that is used for syntax highlighting when using the Contao "code" content element) gets installed by default with a lot of different stylesheets ("Themes") to be able to change the overall colors according to different use cases (I don't mean colors to highlight different programming languages, but overall color variants):

![css-files](https://github.com/contao/contao/assets/31544192/c2b3163d-6f44-4248-beae-764aafdcc08c)

You can see and test the different "Themes" here: https://highlightjs.org/examples

Currently the use of **foundation.css** is set as default directly inside the Twig template of the code content element.

I know you can overwrite it by using customized Twig templates. But I thought it would be handy to have a dropdown selection field per code content element in the backend.

This would make it possible for editors/non-admins to easily select different highlight CSS variants and have the option to use different themes in parallel.

I think **foundation.css** could still be declared as standard and also be used as fallback (when the backend user didn't choose another specific Stylesheet at the code element itself).

**Comment by fritzmg** [comment]
> This would make it possible for editors/non-admins to easily select different highlight CSS variants and have the option to use different themes in parallel.

Personally I don't see the use case. Typically whoever is in charge will want to dictate which design should be used for a certain theme and it should not be possible for editors to change this. But then again, in such a case you can easily just not unlock this future `tl_content` field in the group settings 👍

**Comment by leofeyer** [comment]
I don‘t think it makes sense to use a different style for each code element. Normally, the designer chooses the style that best suits the design of the website and this should then be used for all code elements. And to me, the template seems the perfect place to make this choice.

**Comment by m-vo** [comment]
If you want to have different configurations, you could crate variant templates like so and then select them in the back end:

```twig
{# content_element/code/whatever.html.twig #}
{% extends "@Contao/content_element/code.html.twig" %}

{% block style %}
    {% set highlighter_css_file = asset('whatever.css', 'scrivo/highlight.php') %}
    {{ parent() }}
{% endblock %}
```

**Comment by CMSworker** [comment]
Thank you @m-vo, I didn't think about that. This is basically kind of what I requested/suggested (dropdown to select a style). Only difference is that selectable styles/templates have to be defined by the Admin.

**Unfortunatelly it just doesn't work this way at the moment:** Only the CSS file of the template selected in the last added code element gets loaded and will be used for all code elements on this page. Because "set highlighter_css_file" overwrites the previous definitions of the other code templates used on the same page. Should I open a new issue for that problem?

I think this feature request could be closed then. But I'd like to suggest to add a hint somewhere that it is possible to define a different highlight CSS files and (once it is possible) how to use different stylings for code elements.

If you prefere to add a note in the documentation I could help adding it. I'm just not sure, where... at [Inhaltselemente](https://docs.contao.org/manual/de/artikelverwaltung/inhaltselemente/text-elemente/#code)? (there are no other examples how to customize templates, so probably not there)

I personally think the note would be placed best in the [template skeleton](https://github.com/contao/contao/blob/5.x/core-bundle/contao/templates/twig/backend/template_skeleton.html.twig) comment block, but I don't know how to add it in a way that it will only be added to the code element templates.

// Edit: Just wanted to add another use case: To use different hightlighter styles for dark mode and light mode...

**Comment by m-vo** [comment]
> Only the CSS file of the template selected in the last added code element gets loaded and will be used for all code elements on this page

If you need different styles on a single page, you need to ship your own CSS anyways. The default highlight.php themes all got the same selectors.

**Comment by CMSworker** [comment]
Closing this for now.

In case someone finds this, searching for how to change the default code highlight CSS in Contao 5:

Add a new twig template [content_element/code.html.twig]

To change the default highlighter_css_file, add the following code block and change the name of the CSS file to any other css file that is available in /vendor/scrivo/highlight.php/styles/:

```php
{% block style %}
  {% set highlighter_css_file = asset('foundation.css', 'scrivo/highlight.php') %}
  {{ parent() }}
{% endblock %}
```

To use your own css file, change the link to a file in your /files/ folder:

```php
{% block style %}
  {% set highlighter_css_file = '/files/css/custom-highlighter.css' %}
  {{ parent() }}
{% endblock %}
```
Don't forget to clear/rebuild the production cache after changes in the twig templates.

As a starting base for your own CSS file:

Copy the content of any CSS file within /vendor/scrivo/highlight.php/styles/ and change it to your needs.

Within your own css file, you can

- define styles for dark mode / light mode, for example with `@media (prefers-color-scheme: dark) {}`
- duplicate css elements and add different classes before them, to be able to used different stylings for different code elements
